prov/sockets/src/sock_ep_msg.c

```diff
@@ -693,8 +694,10 @@ static void *sock_msg_ep_listener_thread(void *data)
 	poll_fds[0].events = poll_fds[1].events = POLLIN;
 
 	while (*((volatile int*) &ep->attr->cm.do_listen)) {
+		fastlock_acquire(&ep->attr->cm.lock);
 		timeout = dlist_empty(&ep->attr->cm.msg_list) ? -1 :
 					SOCK_CM_COMM_TIMEOUT;
+		fastlock_release(&ep->attr->cm.lock);
```

Comment:
What happens if the item is removed from the list immediately after the lock is released?